lib/AppleNotesSmartFolder.rb

```diff
@@ -31,10 +31,21 @@ def to_csv
   end
 
   def generate_html
-    html = "<a id='folder_#{@primary_key}'><h1>#{@account.name} - #{full_name}</h1></a>"
-    html += "A smart folder looking for notes matching: <code>#{query}</code>"
-
-    return html
+    builder = Nokogiri::HTML::Builder.new(encoding: "utf-8") do |doc|
+      doc.h1 {
```

Comment:
Based on looking at the other classes, I suspect this `doc.h1` and `doc.div` should be nested within a larger `doc.div` root node. This class generates an error "A document may not have multiple root nodes." for this function.